docs/build/cookbook.rst

```diff
@@ -1573,4 +1573,41 @@ the same ``env.py`` file can be invoked using asyncio as::
             await conn.run_sync(run_upgrade, config.Config("alembic.ini"))
 
 
-    asyncio.run(run_async_upgrade())
\ No newline at end of file
+    asyncio.run(run_async_upgrade())
+
+
+Data migrations
+===============
+
+Alembic migrations are designed for schema migrations.
+The nature of data migrations are inherently different and it's not in fact advisable in the general case to write data migrations that integrate with Alembic's schema versioning model.
+For example downgrades are difficult to address since they might require deletion of data, which may even not be possible to detect.
+
```

Comment:
**Federico Caselli** (CaselIT) wrote:

Ack

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/3895